### DISCUSSION TITLE
Multi-path Celestia Namespace Utilization

### DISCUSSION THREAD

**Discussion by l-monninger** [movementlabsxyz/movement#345] [Ideas]
# Summary
There may be a way to increase Celestia throughput without an additional data dissemination layer. This approach could in theory be applied to other platforms with similar namespace approaches and confirmation-based approaches, e.g., Avail. 

Data dissmenination would allow for greater throughput and lower gas fees. But, this approach is worthy note. 

## Multipath Namespaces
![Screenshot 2024-08-13 at 12 16 16 PM](https://github.com/user-attachments/assets/3f3f5e6c-83cd-4716-8228-353cc83eb7b5)

The basic idea is:
1. Use a finite set of $n$ namespaces.
2. Define a rule that determines the order between these namespaces.
3. Rather than waiting for confirmation on a blob submission for a namespace before sending the next, send blob submissions to fill out up to $n$ namespaces.
4. On receipt of confirmation, ignore all remaining blobs based on the computable order after any such blob submissions request which fails. 

This is conceptually similar to OFDM channel usage with sequence numbers. Hence this I may refer to these as channels in this discussion.

**Comment by musitdev** [comment]
In the Celestia, I saw this remark: https://docs.celestia.org/developers/submit-data#submitting-multiple-transactions-in-one-block-from-the-same-account
As I understand it, you can send several blob Tx without waiting to be added to a block. Perhaps it has the same role, not waiting between the two propositions.

**Reply by l-monninger** [reply]
Sorry, I had a response typed and forgot to send it. 

We can give this a try. We will need to learn how to construct a `BlobTx` appropriately. I don't know if this is super ergonomic in Lumina.
